### PR TITLE
ISPN-5380 Internal Cache Registry

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-5380
https://issues.jboss.org/browse/ISPN-5381
https://issues.jboss.org/browse/ISPN-5382

### CODE REVIEW

File: core/src/main/java/org/infinispan/registry/impl/InternalCacheRegistryImpl.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * InternalCacheRegistryImpl.
+ *
+ * @author Tristan Tarrant
+ * @since 7.2
+ */
+public class InternalCacheRegistryImpl implements InternalCacheRegistry {
+   private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+   private EmbeddedCacheManager cacheManager;
+   private final Set<String> internalCaches = new HashSet<>();
```
Comment (gustavonalle) [inline core/src/main/java/org/infinispan/registry/impl/InternalCacheRegistryImpl.java]: Does it need a concurrent collection here?

Comment (tristantarrant) [inline core/src/main/java/org/infinispan/registry/impl/InternalCacheRegistryImpl.java]: Yes, better stay on the safe side. I guess I can just use a ConcurrentHashMap<String, Boolean> where the value is meaningless

Comment (gustavonalle) [inline core/src/main/java/org/infinispan/registry/impl/InternalCacheRegistryImpl.java]: There's a ```ConcurrentHashSet``` internally you could use as well

**Comment by tristantarrant** [comment]
@gustavonalle changed it

**Comment by tristantarrant** [comment]
Dropped ISPN-5383 commit and issued a new PR just for that

**Comment by gustavonalle** [comment]
pulling

**Comment by anistor** [comment]
@tristantarrant @gustavonalle Shouldn't we use the internal cache registry for the QueryCache too?

**Comment by gustavonalle** [comment]
I'm picking some failures on ```client/hotrod-client```:

```org.infinispan.server.hotrod.CacheNotFoundException: Cache with name '___script_cache' not found amongst the configured caches```

```org.infinispan.server.hotrod.CacheNotFoundException: Cache with name '___protobuf_metadata' not found amongst the configured caches```

do you see them as well?

**Comment by tristantarrant** [comment]
Let me check

On Fri, Apr 17, 2015 at 1:05 PM, Gustavo Fernandes <notifications@github.com

**Comment by anistor** [comment]
I get the same errors, with or without rebasing this to latest.

**Comment by tristantarrant** [comment]
I added the concept of "USER" caches, which are internal caches which are user-visible. And I've also moved the logic of filtering out internal caches to the registry itself.

**Comment by gustavonalle** [comment]
@anistor the QueryCache could use this too. Do you see any issue of doing it in a separate jira?

**Comment by anistor** [comment]
@gustavonalle I'll add a jira for using the internal cache registry for QueryCache

**Comment by anistor** [comment]
@gustavonalle Just added an issue to further track it: https://issues.jboss.org/browse/ISPN-5390

**Comment by gustavonalle** [comment]
merged. tks @tristantarrant

**Pull request closed without merging**